fvm/environment/derived_data_invalidator.go

```diff
@@ -22,10 +22,42 @@ type ContractUpdate struct {
 type DerivedDataInvalidator struct {
 	ContractUpdateKeys []ContractUpdateKey
 	FrozenAccounts     []common.Address
+
+	MeterParamOverridesUpdated bool
 }
 
 var _ programs.TransactionInvalidator = DerivedDataInvalidator{}
 
+func NewDerivedDataInvalidator(
+	contractKeys []ContractUpdateKey,
+	env *facadeEnvironment,
+) DerivedDataInvalidator {
+	return DerivedDataInvalidator{
+		ContractUpdateKeys:         contractKeys,
+		FrozenAccounts:             env.FrozenAccounts(),
+		MeterParamOverridesUpdated: meterParamOverridesUpdated(env),
+	}
+}
+
+func meterParamOverridesUpdated(env *facadeEnvironment) bool {
+	updatedRegisterIds, _ := env.txnState.RegisterUpdates()
+
+	serviceAccount := string(env.chain.ServiceAddress().Bytes())
+	storageDomain := common.PathDomainStorage.Identifier()
+	for _, registerId := range updatedRegisterIds {
+		if registerId.Owner != serviceAccount {
+			continue
+		}
+
+		if registerId.Key == storageDomain ||
+			state.IsStabIndex(registerId.Key) {
```

Comment:
Can you please add a comment why we need to invalidate meter params in these two cases?